### ISSUE TITLE
Incorrect entry in channellog file after echoerr test

### ISSUE THREAD

**Issue by ghost** [vim/vim#8686]
**Describe the bug**
When the "echoerr" test in function Ch_communicate() in the file testdir/test_channel.vim is run, it generates an exception which is caught as it should be.  However, that test then writes an incorrect error message in the "channellog" file as shown below in **To Reproduce**.

**To Reproduce**
1.  cd src/testdir
2.  Enable the channellog in test_channel.vim (uncomment "call ch_logfile" on line 20)
3.  run the test:  make test_channel.res
4.  all tests should pass
5.  view the channellog file:  vim +1 -c /E491 -c /   channellog

If the bug is present, you should see something like this:

  3.324037 on 35: Executing ex command 'echoerr "this is an error"'
  3.324137 on 35: Ex command error: 'E491: json decode error at 'wrong!["ex","smi"]''

The error message is apparently from the prior "malformed1" test which, if you jump back to its test results, has generated these entries:

  4.260544 RECV on 37(sock): '["ex",":"]wrong!["ex","smi"]'
  4.260620 : ERROR silent: E491: json decode error at 'wrong!["ex","smi"]'
  4.260656 ERR on 37: Decoding failed - discarding input

**Expected behavior**
I'm not sure but what it does do is misleading at best.  An exception has been generated and caught, v:errmsg has not been set but the Vim code that executes the command chooses to print the value of v:errmsg.

**Screenshots**
N/A

**Environment**
 - Vim version 8.2.3242, normal features, no gui
 - OS: debian linux
 - Terminal: xterm-256

**Additional comments**
If v:errmsg is cleared (set to empty string) just before the echoerr test, these lines are written to channellog:

  4.891479 on 37: Executing ex command 'echoerr "this is an error"'
  4.891583 on 37: Ex command error: ''

**Comment by brammool** [comment]
I don't think it is worth spending time on this.  The channel stuff is asynchronous by nature, the log can show messages out of sequence.

**Comment by msoyka-of-wharton** [comment]
Bram,

From your response I think I haven't clearly explained the problem as I see it.  Let me try to clarify my concern.  

The problem is not that something is written to the channellog out of sequence.  The message I referred to, E491, actually appears twice- once where it should be as part of the "malformed1" test and a second time where it shouldn't be, apparently as part of the "echoerr" test.  I've also shown (see "Additional comments") that the second occurrence is not due to some delayed handling because when I clear v:errmsg before the "echoerr" test, the error message printed in channellog is empty.  

So my concern was that Vim appears to think that the "echoerr" test has generated an error condition and prints the value of v:errmsg but nothing is setting the v:errmsg variable.  I don't claim to know what the expected behavior is with regards to v:errmsg when an exception is generated and caught but, what I'm seeing does not meet my expectations, hence this bug report.  

-mike
